src/TypeSpec.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -21,13 +21,38 @@ import {
     resolveAzureEmitterOptions
 } from "./options.js";
 import { azureSDKContextOptions } from "./sdk-context-options.js";
+import { getArmResources } from "@azure-tools/typespec-azure-resource-manager";
 
 export async function $onEmit(context: EmitContext<AzureNetEmitterOptions>) {
     const program: Program = context.program;
     const options = resolveAzureEmitterOptions(context);
     /* set the loglevel. */
     Logger.initialize(program, options.logLevel ?? LoggerLevel.INFO);
 
+    const details = getArmResources(program);
```

Comment:
maybe we could put [this](https://github.com/microsoft/typespec/issues/4342) into our agenda and do it first, so that we could have the full codeModel available right here, and we do not have to emit another file, but we could emit an extra section in the tspCodeModel.json